.azure-pipelines.yml

```
@@ -96,14 +95,16 @@ jobs:
 
   - script: |
       python -m pip install --upgrade pip
-      pip install setuptools setuptools_scm pytoml wheel twine
-    displayName: 'Install build dependencies'
+      pip install flit twine wheel pytoml
+      python -c \
+        'import pytoml; c=pytoml.load(open("pyproject.toml")); print("\0".join(c["build-system"]["requires"]), end="")' \
+        | xargs -0 pip install
```

```suggestion
      pip install flit twine wheel pytoml setuptools_scm
```

I think this is going to be easier to figure out.

I take it `flit` isn't using the `build-system.requires` information by itself?

---

It is for `flit install`. Remember that `flit build` isn’t usually needed, it’s effectively used as a debug tool to inspect packages in a dev environment.